preliz/ppls/pymc_io.py

```diff
@@ -210,12 +211,40 @@ def non_constant_parents(var_, free_rvs):
     return parents
 
 
-def parse_backfitting(model, data):
-    posterior = az.extract(data, group="posterior")
+def parse_backfitting(model, data, alternative=None):
+    """
+    Get the model information for fitting the samples from prior into user provided model's prior.
+    We need to "backfit" because we can not use arbitrary samples as priors.
 
-    model_info = get_model_information(model)[2]
-    parsed_info = [(dist, var) for var, dist in model_info.items()]
+    Parameters
+    ----------
+    model : A PyMC model
+    A probabilistic model
+
+    data : np.ndarray
+    Samples which needs to be fitted using "backfitting"
+
+    alternative : list, defaults to None
```

Comment:
probably a very flexible solution will be to make this accept None, "auto", list or dict

* If None only evaluate original family
* If "auto" evaluate original + a set of default distributions
* If list evaluate original + the distributions inside the list. The list should be valid preliz distributions
* If dict evaluate original + the values in the dict. The keys should be var_names in the model and the values valid preliz distributions.

Reply:
Have made the changes for the 4 options,
For the 2,3 and 4th we take the term with the least penalization term for each variable